pyomo/gdp/plugins/bound_pretransformation.py

```diff
@@ -230,77 +233,99 @@ def _update_bounds_dict(self, v_bounds, lower, upper, disjunct, gdp_forest):
-                        if need_lb and lb is not None:
-                            lb_expr += lb * indicator_var
-                            need_lb = False
-                        if need_ub and ub is not None:
-                            ub_expr += ub * indicator_var
-                            need_ub = False
-                    if disjunct is None:
-                        break
-                    disjunct = gdp_forest.parent_disjunct(disjunct)
-                if need_lb:
-                    all_lbs = False
-                if need_ub:
-                    all_ubs = False
-            deactivate_lower = set()
-            deactivate_upper = set()
-            if all_lbs:
-                idx = (v.local_name + '_lb', unique_id)
-                trans_block.transformed_bound_constraints[idx] = lb_expr <= v
-                trans_map[v] = [trans_block.transformed_bound_constraints[idx]]
-                for c in v_bounds['to_deactivate']:
-                    if c.upper is None:
-                        c.deactivate()
-                    elif c.lower is not None:
-                        deactivate_lower.add(c)
-                disjunction._transformation_map
-            if all_ubs:
-                idx = (v.local_name + '_ub', unique_id + 1)
-                trans_block.transformed_bound_constraints[idx] = ub_expr >= v
-                if v in trans_map:
-                    trans_map[v].append(trans_block.transformed_bound_constraints[idx])
-                else:
+        to_transform = ComponentSet([root_disjunction])
+
+        while to_transform:
+            disjunction = to_transform.pop()
+
+            trans_block = self._add_transformation_block(
+                disjunction, transformation_blocks
+            )
+            if self.transformation_name not in disjunction._transformation_map:
+                disjunction._transformation_map[
+                    self.transformation_name
+                ] = ComponentMap()
+            trans_map = disjunction._transformation_map[self.transformation_name]
+
+            for disj in disjunction.disjuncts:
+                to_transform.update(gdp_forest.children(disj))
+
+            for v, v_bounds in bound_dict.items():
+                unique_id = len(trans_block.transformed_bound_constraints)
+                if not any(disj in v_bounds for disj in disjunction.disjuncts):
+                    continue
+                all_lbs = True
+                all_ubs = True
+                lb_expr = 0
+                ub_expr = 0
+                deactivate_lower = set()
+                deactivate_upper = set()
+                for disj in disjunction.disjuncts:
+                    (lb, ub) = self._get_tightest_ancestral_bounds(
+                        v_bounds, disj, gdp_forest
+                    )
+                    if lb is None:
+                        all_lbs = False
+                    if ub is None:
+                        all_ubs = False
```

Comment:
Is it worth breaking out of this loop early if both `all_lbs` and `all_ubs` are False?  Something like
```python
if lb is None:
    all_lbs = False
    if not all_ubs:
        break
if ub is None:
    all_ubs = False
    if not all_lbs:
        break
```

Reply:
Oh, it probably is. In the absence of performance tests, I'm going to do it for now.